drivers/wifi/eswifi/eswifi_bus_spi.c

```diff
@@ -0,0 +1,234 @@
+/**
+ * Copyright (c) 2018 Linaro
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_WIFI_LEVEL
+#define SYS_LOG_DOMAIN "dev/eswifi"
+#if (SYS_LOG_LEVEL > SYS_LOG_LEVEL_OFF)
+#define NET_LOG_ENABLED 1
+#endif
+
+#include <logging/sys_log.h>
+
+#include <zephyr.h>
+#include <kernel.h>
+#include <device.h>
+#include <string.h>
+#include <errno.h>
+#include <gpio.h>
+#include <spi.h>
+
+#include "eswifi.h"
+
+#define ESWIFI_SPI_THREAD_STACK_SIZE 1024
+K_THREAD_STACK_MEMBER(eswifi_spi_poll_stack, ESWIFI_SPI_THREAD_STACK_SIZE);
+
+struct eswifi_spi_data {
```

Comment:
this specific file is really for SPI and this structure is specific to SPI bus, so it makes sense to me to name it _spi. If a new bus support (USB, UART, I2c...) is implemented, it will go to a separate file with it's own specific bus data structures. the core layer is the 'generic' part and defines a eswifi_device struct.

Reply:
I guess you mean that it is meant to access spi based eswifi. Since the only spi related part here is the spi dev pointer (from zephyr point of view). The gpio and k_thread stuff are just generic.
It feels awkward to see in later code spi->dr.dev for instance. dr being a gpio line used for notification and which has nothing to do with the spi bus itself. It's confusing.

Reply:
Yes eswifi_spi_data is just to be aligned with the context and naming convention of the file, this is a sublayer of eswifi dedicated to spi communication, the thread, data-ready gpios... being only used for spi mode (for polling).